x/concentrated-liquidity/types/keys.go

```diff
@@ -90,11 +90,15 @@ func KeyPool(poolId uint64) []byte {
 	return []byte(fmt.Sprintf("%s%d", PoolPrefix, poolId))
 }
 
-func KeyIncentiveRecord(poolId uint64, denom string, minUptime time.Duration, addr sdk.AccAddress) []byte {
+func KeyIncentiveRecord(poolId uint64, minUptimeIndex int, denom string, addr sdk.AccAddress) []byte {
 	addrKey := address.MustLengthPrefix(addr.Bytes())
-	return []byte(fmt.Sprintf("%s%s%d%s%s%s%d%s%s", IncentivePrefix, KeySeparator, poolId, KeySeparator, denom, KeySeparator, uint64(minUptime), KeySeparator, addrKey))
+	return []byte(fmt.Sprintf("%s%s%d%s%d%s%s%s%s", IncentivePrefix, KeySeparator, poolId, KeySeparator, minUptimeIndex, KeySeparator, denom, KeySeparator, addrKey))
```

Comment:
Mainly since when we serialize minUptimeValue into bytes, the store iterator seems to not be able to tell the difference between the full number and its first two bytes (as a toy example to demonstrate, it might only read the first two characters of `10,000` and therefore conclude that `110 > 10,000` since `11 > 10`).

Using the index lets us get around this because we are guaranteed to have a small number of supported indices (specifically, a lower number than what we can fit into two bytes). It's likely possible to do a workaround to make it work with `minUptimeValue`, but this approach seemed both simpler and in line with our convention of using IDs in keys.